fdbserver/DataDistribution.actor.cpp

```diff
@@ -1335,6 +1343,140 @@ ACTOR Future<Void> bulkLoadingCore(Reference<DataDistributor> self, Future<Void>
 	}
 }
 
+ACTOR Future<Void> doBulkDumpTask(Reference<DataDistributor> self,
+                                  StorageServerInterface ssi,
+                                  BulkDumpState bulkDumpState,
+                                  std::vector<UID> checksumServers) {
+	TraceEvent(SevInfo, "DDBulkDumpDoTask", self->ddId).detail("BulkDumpState", bulkDumpState.toString());
+	try {
+		ErrorOr<BulkDumpState> vResult =
+		    wait(ssi.bulkdump.tryGetReply(BulkDumpRequest(checksumServers, bulkDumpState)));
+		if (vResult.isError()) {
+			throw vResult.getError();
+		}
+	} catch (Error& e) {
+		if (e.code() == error_code_actor_cancelled) {
+			throw e;
+		}
+	}
+	self->bulkDumpParallelismLimitor.decrementTaskCounter();
+	return Void();
+}
+
+ACTOR Future<bool> scheduleBulkDumpTasks(Reference<DataDistributor> self) {
```

Comment:
This schedules all bulkdump tasks for the whole user keyrange?

Reply:
No. It reads the metadata (it read all user ranges at this part) and check where part range is set (by users) to do bulkdump.